pkg/agent/tcpconn/disruptor.go

```diff
@@ -0,0 +1,108 @@
+// Package tcpconn contains a TCP connection disruptor.
+package tcpconn
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/grafana/xk6-disruptor/pkg/iptables"
+	"github.com/grafana/xk6-disruptor/pkg/runtime"
+)
+
+// Disruptor applies TCP Connection disruptions.
+type Disruptor struct {
+	Executor   runtime.Executor
+	Queue      Queue
+	Disruption Disruption
+	NFQConfig  NFQConfig
+}
+
+// Disruption holds the parameters that describe a TCP connection disruption.
+type Disruption struct {
+	// Port is the target port to match which connections will be intercepted.
+	Port uint
+	// DropRate is the rate in [0, 1] range of connections that should be dropped.
+	DropRate float64
+	// DropPeriod defines how often connections will be terminated according to DropRate.
+	// If set to zero, connections will only be terminated once. If it is set to a non-zero value, connections will be
+	// re-evaluated for termination when this period passes.
+	DropPeriod time.Duration
+}
+
+// ErrDurationTooShort is returned when the supplied duration is smaller than 1s.
+var ErrDurationTooShort = errors.New("duration must be at least 1 second")
+
+// Apply executes the configured disruption for the specified duration.
+func (d Disruptor) Apply(ctx context.Context, duration time.Duration) error {
+	if duration < time.Second {
+		return ErrDurationTooShort
+	}
+
+	iptables := iptables.New(d.Executor)
+	//nolint:errcheck // Nothing to do while we don't implement logging.
+	defer iptables.Remove()
```

Comment:
In hindsight, the responsibility of setting up iptables rules is probably better placed in the queue. I will probably refactor this soon.